src/CalendarStrip.js

```diff
@@ -146,23 +146,17 @@ class CalendarStrip extends Component {
     // Animate showing of CalendarDay elements
     this.animate();
 
-    let selectedDate = {},
-      startingDate = {},
-      weekData = {};
-      
-    selectedDate = {
-      selectedDate: this.setLocale(moment(this.props.selectedDate))
-    };
-    startingDate = {
-      startingDate: this.updateWeekStart(selectedDate.selectedDate)
-    };
-    weekData = this.updateWeekData(
-      startingDate.startingDate,
-      selectedDate.selectedDate,
-      this.props
-    );
-
-    this.setState({...weekData});
+    let selectedDate = this.setLocale(moment(this.props.selectedDate));
```

Comment:
`moment(this.props.selectedDate)` will still create today's date if `props.selectedDate` isn't provided.  The subsequent calculations of `daysFromStart` and `startOfWeek` are based on that.
It looks like the only way to solve this is to check whether `props.selectedDate` is passed in:
```
if (this.props.selectedDate) {
  // Your new code
}
```


(sorry this took a while to respond to... didn't see it initially)

Reply:
Upon reviewing the constructor, it's also doing `moment(this.props.selectedDate)`.  Can't remember why that works.

Is there a good reason your code is in `componentDidMount` instead of the constructor?